### DISCUSSION TITLE
[Symfony 7.1.5] UrlGenerator adds host when using UrlGeneratorInterface::ABSOLUTE_PATH

### DISCUSSION THREAD

**Discussion by smhg** [symfony/symfony#58570] [Q&A]
I think this might be a bug/feature request, but I'd like to double check.

When generating a URL like this:
```php
$url = $this->generateUrl(
    'my_route,
    [],
    UrlGeneratorInterface::ABSOLUTE_PATH
);
```
it will still add the host if the host of `my_route` is different from the host of the current request (e.g. when working with multiple subdomains) due to [this](https://github.com/symfony/symfony/blob/9d89e05279631430f9037844fc4cc2eaac79e886/src/Symfony/Component/Routing/Generator/UrlGenerator.php#L233). It's a bit awkward when specifying ABSOLUTE_PATH, no? But it might still be a good default.

However, when both the route and the current request use a different _port_, it does not add the host:port combination. This causes prod (domain) and dev (localhost:port) urls to be different. That doesn't sound great I think?

Should I file this as an issue?

**Comment by stof** [comment]
If the route requires a specific host and the current host does not match, generating a url containing the request host is expected. This is what allows host requirements to work seamlessly during route generation (throwing an exception telling you the host mismatches would not be great DX).

Can you provide more details about the issue you have with different ports ? How does your route definition and your current request look like ?

**Comment by smhg** [comment]
> If the route requires a specific host and the current host does not match, generating a url containing the request host is expected. 

I agree that check makes a lot of sense. Although, even when `UrlGeneratorInterface::ABSOLUTE_PATH` is used? That parameter is like a guideline? It might still return more than only the path.

On my dev env I use different ports to mimic different subdomains. So in the above scenario when the 'generating' route is passing through one port and the 'generated' route passing through another, this is not picked up by the host check (as it only checks the host). So in dev the hosts are the same (localhost) and this does not add the host to the generated url. I only noticed in production the host was being added. That felt a bit wrong?
